doc/designs/quic-design/connection-id-cache.md

```diff
@@ -0,0 +1,184 @@
+QUIC Connection ID Cache
+========================
+
+Connection ID Cache
+-------------------
+
+A table matching Connection IDs with Connection objects represented
+via SSL objects.
+
+_In MVP there is a many-to-1 matching of Connection IDs to Connection
+objects.  Refer third paragraph in [5.1]_
+
+API
+---
+
+```c
+#define QUIC_MAX_ID_LENGTH  20
+
+struct quic_connection_st;
+struct quic_conn_id_cache_st;
+struct id_elem_st;
+
+typedef struct quic_conn_id_cache_st QUIC_CONN_ID_CACHE;
+typedef struct id_elem_st ID_ELEM;
+
+/*
+ * Allocate and free a connection ID cache
+ */
+QUIC_CONN_ID_CACHE *ossl_quic_new_conn_id_cache(void);
+void ossl_quic_free_conn_id_cache(QUIC_CONN_ID_CACHE *cache);
+
+/*
+ * Add an ID to connection mapping to the cache.
+ */
+int ossl_quic_set1_conn_id(QUIC_CONN_ID_CACHE *cache, int local,
```

Comment:
Calling this `set1` is pretty confusing and doesn't match `remove`. How about `add` or `add1`?

Please discuss how the `local` parameter works more on all functions.

I am slightly skeptical about having these be the same function rather than separate `local` and `remote` functions but this is a matter of taste.

Reply:
As these functions get more complicated, it saves duplicating ever larger amounts of code.

It would be possible to have external APIs for local and remote which call common internal code but it seemed easier to have a flag argument rather than passing more augments around.

Reply:
Yeah. I was thinking of the latter idea. But this is an internal API so I don't mind either way.

Reply:
Changed over, I prefer not having a mystery argument.  Register shuffling is cheap so the rearrangement of the argument lists shouldn't hurt.